src/ledger/LedgerCloseMetaStream.cpp

```diff
@@ -0,0 +1,120 @@
+// Copyright 2019 Stellar Development Foundation and contributors. Licensed
+// under the Apache License, Version 2.0. See the COPYING file at the root
+// of this distribution or at http://www.apache.org/licenses/LICENSE-2.0
+
+#include "ledger/LedgerCloseMetaStream.h"
+#include "util/GlobalChecks.h"
+#include "util/Logging.h"
+#include "xdrpp/marshal.h"
+
+namespace stellar
+{
+
+LedgerCloseMetaStream::LedgerCloseMetaStream(asio::io_context& ctx,
```

Comment:
Do we gain anything from the extra level of buffering to write to the pipe?

For the "rebuild the meta" scenario, the "dumb" implementation would just synchronously write on the main thread (blocking). This would not require any throttling logic, would not do an extra copy of the data into buffers, and the only assumption would be that the reader would consume data as it gets pushed onto the pipe (and as we copy less, we may have a net perf increase).

If it's to make the "online" version work, it's probably premature as this design doesn't handle failures anyways (and this PR doesn't even deal with shutdown properly if not run with `catchup`).

Reply:
Done.